modules/3d/src/usac/estimator.cpp

```diff
@@ -534,6 +602,191 @@ Ptr<ReprojectionErrorPmatrix> ReprojectionErrorPmatrix::create(const Mat &points
     return makePtr<ReprojectionErrorPmatrixImpl>(points);
 }
 
+class PlaneModelErrorImpl : public PlaneModelError, public PointCloudWrapper {
+private:
+    //! ax + by + cz + d = 0
+    float a, b, c ,d;
+    std::vector<float> errors_cache;
+    bool cache_valid;
+
+public:
+    explicit PlaneModelErrorImpl(const Mat &points_)
+    : PointCloudWrapper(points_),
+    a(0), b(0), c(0), d(0), errors_cache(pts_cnt), cache_valid(false)
+    {
+    }
+
+    inline void setModelParameters(const Mat &model) override
+    {
+        CV_Assert(!model.empty());
+        CV_CheckTypeEQ(model.depth(), CV_64F, "Model parameter type should use double");
+
+        const double * const p = (double *) model.data;
+        double coeff_a = p[0], coeff_b = p[1], coeff_c = p[2], coeff_d = p[3];
+        // Format for easy distance calculation
+        double magnitude_abc = sqrt(coeff_a * coeff_a + coeff_b * coeff_b + coeff_c * coeff_c);
+        if (0 != magnitude_abc) {
+            double inv_magnitude_abc = 1.0 / magnitude_abc;
+            coeff_a = coeff_a * inv_magnitude_abc;
+            coeff_b = coeff_b * inv_magnitude_abc;
+            coeff_c = coeff_c * inv_magnitude_abc;
+            coeff_d = coeff_d * inv_magnitude_abc;
+        }
+        cache_valid = cache_valid && a == (float) coeff_a && b == (float) coeff_b
+                &&  c == (float) coeff_c && d == (float) coeff_d;
+
+        a = (float) coeff_a;
+        b = (float) coeff_b;
+        c = (float) coeff_c;
+        d = (float) coeff_d;
+    }
+
+    inline float getError(int point_idx) const override
+    {
+        float error = a * pts_ptr_x[point_idx] + b * pts_ptr_y[point_idx]
+                      + c * pts_ptr_z[point_idx] + d;
+        return error * error;
+    }
+
+    const std::vector<float> &getErrors(const Mat &model) override
+    {
+        setModelParameters(model);
+
+        if (cache_valid)
+            return errors_cache;
+
+        int i = 0;
+#ifdef CV_SIMD
+        int end = (pts_cnt / v_float32::nlanes) * v_float32::nlanes;
+
+        v_float32 v_a = vx_setall_f32(a);
+        v_float32 v_b = vx_setall_f32(b);
+        v_float32 v_c = vx_setall_f32(c);
+        v_float32 v_d = vx_setall_f32(d);
+
+        float* errors_cache_ptr = errors_cache.data();
+        for (; i < end; i += v_float32::nlanes)
```

Comment:
```.diff
-i < end
+i <= pts_cnt - v_float32::nlanes
```

To keep condition and increment on the same line (the same note is about all OpenCV SIMD loops)

Reply:
fixed.